src/biogeophys/OzoneMod.F90

```diff
@@ -472,6 +425,108 @@ subroutine CalcOzoneStressOnePoint( &
        o3uptake = 0._r8
     end if
 
+  end subroutine CalcOzoneUptakeOnePoint
+
+  !-----------------------------------------------------------------------
+  subroutine CalcOzoneStress(this, bounds, num_exposedvegp, filter_exposedvegp)
+    !
+    ! !DESCRIPTION:
+    ! Calculate ozone stress.
+    !
+    ! !ARGUMENTS:
+    class(ozone_type), intent(inout) :: this
+    type(bounds_type), intent(in) :: bounds
+    integer  , intent(in) :: num_exposedvegp           ! number of points in filter_exposedvegp
+    integer  , intent(in) :: filter_exposedvegp(:)     ! patch filter for non-snow-covered veg
+    !
+    ! !LOCAL VARIABLES:
+
+    character(len=*), parameter :: subname = 'CalcOzoneStress'
+    !-----------------------------------------------------------------------
+
+    select case (this%stress_method)
+    case (stress_method_lombardozzi)
+       call this%CalcOzoneStressLombardozzi(bounds, num_exposedvegp, filter_exposedvegp)
```

Comment:
Here is where this will be extended to allow the Luna-specific method from @ziu1986. I envision that CalcOzoneStress will be called from the same place in the code in either case (currently clm_driver.F90) - i.e., the caller won't need to know which ozone stress method is being used.

It looks like the Luna-based method only needs to calculate the stress coefficient once a day, at the end of the day. So, to give a small performance improvement, we can put an end-of-day check either in the case block (as a conditional around the call to the new routine) or at the top of the new routine (simply returning if it isn't the end of the current day).

In terms of the calculation of the different stress coefficients (of which there will be 6): My thinking is that we can rely on the fact that they are all initialized to 1 in InitCold. So the new o3coefjmax will *not* be set explicitly in CalcOzoneStressLombardozzi, and similarly, the original o3coefv and o3coefg variables will *not* be set explicitly in your new method.

Reply:
Am I right to assume that instead of storing the O3 coefficients we are going to store the O3 uptake in the history/restart files? Not sure if I have seen that in the proposed changes.

Reply:
The o3 uptake was already in the restart files. This is needed because it is an evolving state variable. Previously, we also stored the o3 coefficients in the restart file. That was only needed because of the driver loop ordering: the coefficients were calculated after photosynthesis in time step N and then applied in photosynthesis in time step N+1. I have reordered the code here so that the coefficients are calculated before photosynthesis in time step N, and then applied later that same time step, so they no longer need to be on the restart file.